### DISCUSSION TITLE
Linting and checking validity of types  ...Failed to compile.

### DISCUSSION THREAD

**Discussion by ahmed-elshahat-702** [vercel/next.js#72518] [Help]
### Summary

Type '{ __tag__: "DELETE"; __param_position__: "second"; __param_type__: Promise<{ id: string; }>; }' does not satisfy the constraint 'ParamCheck<RouteContext>'.
  Types of property '__param_type__' are incompatible.
    Property 'params' is missing in type 'Promise<{ id: string; }>' but required in type 'RouteContext'.
![Screenshot 2024-11-08 175301](https://github.com/user-attachments/assets/c05b0e1e-d374-4d71-bb37-cdb2ed8dd540)

### Additional information
import jwt from "jsonwebtoken";
import ListModel from "../../models/ListModel";
import { NextResponse } from "next/server";
import { cookies } from "next/headers";

interface JwtPayload {
  userId: string;
}

export async function DELETE(
  request: Request,
  { params }: { params: { id: string } }
) {
  try {
    const cookiesStore = await cookies();
    const token = cookiesStore.get("token")?.value;

    if (!token) {
      return NextResponse.json(
        { message: "Authentication required" },
        { status: 401 }
      );
    }

    const decoded = jwt.verify(token, process.env.JWT_SECRET!) as JwtPayload;

    // Find list and verify ownership
    const listId = params.id;
    const list = await ListModel.findById(listId);
    if (!list) {
      return NextResponse.json({ message: "list not found" }, { status: 404 });
    }

    if (list.owner.toString() !== decoded.userId) {
      return NextResponse.json(
        { message: "Not authorized to delete this list" },
        { status: 403 }
      );
    }

    // Delete the list
    await ListModel.findByIdAndDelete(listId);

    return NextResponse.json({ message: "list deleted successfully" });
  } catch (error) {
    const errorMessage =
      error instanceof Error ? error.message : "Unknown error";
    return NextResponse.json(
      { message: "Error deleting list", error: errorMessage },
      { status: 500 }
    );
  }
}

_No response_

**Comment by icyJoseph** [comment]
Hi, 

In Next 15, there's a breaking change!

- https://nextjs.org/blog/next-15#async-request-apis-breaking-change

So, now to access, `params`, you have to use a promise:

```ts
export async function DELETE(
request: Request,
{ params }: { params: Promise<{ id: string }> }
) {
// further down

const { id: listId } = await params;
```

See more here: https://nextjs.org/docs/app/building-your-application/upgrading/version-15#temporary-synchronous-usage
